pandas/tests/io/parser/test_header.py

```diff
@@ -684,3 +684,21 @@ def test_header_delim_whitespace(all_parsers):
     result = parser.read_csv(StringIO(data), delim_whitespace=True)
     expected = DataFrame({"a,b": ["1,2", "3,4"]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_usecols_no_header_pyarrow(pyarrow_parser_only):
+    parser = pyarrow_parser_only
+    data = """
+a,i,x
+b,j,y
+"""
+    result = parser.read_csv(
+        StringIO(data),
+        header=None,
+        usecols=[0, 1],
+        dtype="object",
```

Comment:
Can you test with `string[pyarrow]` here

Reply:
Sure things.  It works fine so long as I make the comparison dataframe string[pyarrow] too of course:

```
        StringIO(data),
        header=None,
        usecols=[0, 1],
        dtype="string[pyarrow]",
        dtype_backend="pyarrow",
        engine="pyarrow",
    )
    expected = DataFrame([
        ["a", "i"], ["b", "j"]], 
        dtype="string[pyarrow]"
    )
    tm.assert_frame_equal(result, expected)
```
I'll update the PR with the change.